wp-content/themes/genesis-starter/functions.php

```diff
@@ -37,11 +37,20 @@
 
 //* Setup theme
 \add_action( 'after_setup_theme', function () {
-	$scripts    = new \Genesis_Starter\Scripts();
-	$styles     = new \Genesis_Starter\Styles();
-	$shortcodes = new \Genesis_Starter\Shortcodes();
-
-	$scripts->ready();
-	$styles->ready();
-	$shortcodes->ready();
+	$components = array(
+		'scripts'    => new \Patricia\Scripts(),
+		'shortcodes' => new \Patricia\Shortcodes(),
+		'styles'     => new \Patricia\Styles(),
+	);
+
+	// Excluded components
+	$exclude = \apply_filters( 'gs_exclude_components', array() );
```

Comment:
The idea would be to possibly filter the one or more components in specific pages. With this you don't need any conditions either on the classes themselves or on the pages.

It also allows for the usage of plugins to exclude things.